packages/framework/fluid-framework/src/index.ts

```diff
@@ -22,85 +22,75 @@ export type {
 	LoadableObjectClassRecord,
 	LoadableObjectCtor,
-	LoadableObjectRecord,
 	MemberChangedListener,
 	SharedObjectClass,
 } from "@fluidframework/fluid-static";
+export type { ISharedMap, ISharedMapEvents, IValueChanged } from "@fluidframework/map";
+export { SharedMap } from "@fluidframework/map";
+
 export type {
-	ICreateInfo,
-	IDirectory,
-	IDirectoryClearOperation,
-	IDirectoryCreateSubDirectoryOperation,
-	IDirectoryDataObject,
-	IDirectoryDeleteOperation,
-	IDirectoryDeleteSubDirectoryOperation,
-	IDirectoryEvents,
-	IDirectoryKeyOperation,
-	IDirectoryNewStorageFormat,
-	IDirectoryOperation,
-	IDirectorySetOperation,
-	IDirectoryStorageOperation,
-	IDirectorySubDirectoryOperation,
-	IDirectoryValueChanged,
-	ILocalValue,
-	ISerializableValue,
-	ISerializedValue,
-	ISharedDirectory,
-	ISharedDirectoryEvents,
-	ISharedMap,
-	ISharedMapEvents,
-	IValueChanged,
-} from "@fluidframework/map";
-export {
-	DirectoryFactory,
-	LocalValueMaker,
-	MapFactory,
-	SharedDirectory,
-	SharedMap,
-} from "@fluidframework/map";
-export type {
-	DeserializeCallback,
-	IInterval,
-	IIntervalCollectionEvent,
-	IIntervalHelpers,
-	IJSONRunSegment,
-	IMapMessageLocalMetadata,
-	IIntervalCollection,
-	IntervalLocator,
-	ISequenceDeltaRange,
-	ISerializableInterval,
-	ISerializedInterval,
-	ISharedIntervalCollection,
-	ISharedSegmentSequenceEvents,
-	ISharedString,
-	IValueOpEmitter,
-	SerializedIntervalDelta,
-	SharedStringSegment,
-} from "@fluidframework/sequence";
+	AllowedTypes,
+	ApplyKind,
+	ArrayToUnion,
+	CheckoutEvents,
+	Events,
+	ExtractItemType,
+	FlexList,
+	FlexListToUnion,
+	IDisposable,
+	ImplicitAllowedTypes,
+	ImplicitFieldSchema,
+	InsertableObjectFromSchemaRecord,
+	InsertableTreeFieldFromImplicitField,
+	InsertableTreeNodeFromImplicitAllowedTypes,
+	InsertableTypedNode,
+	IsEvent,
+	ISubscribable,
+	ITree,
+	LazyItem,
+	MakeNominal,
+	NodeBuilderData,
+	NodeFromSchema,
+	ObjectFromSchemaRecord,
+	RestrictiveReadonlyRecord,
+	Revertible,
+	TreeApi,
+	TreeArrayNodeBase,
+	TreeFieldFromImplicitField,
+	TreeLeafValue,
+	TreeMapNode,
+	TreeMapNodeBase,
+	TreeNodeEvents,
+	TreeNodeFromImplicitAllowedTypes,
+	TreeNodeSchema,
+	TreeNodeSchemaClass,
+	TreeNodeSchemaCore,
+	TreeNodeSchemaNonClass,
+	TreeView,
+	Unhydrated,
+	WithType,
+} from "@fluidframework/tree";
 export {
-	getTextAndMarkers,
-	Interval,
-	intervalLocatorFromEndpoint,
-	IntervalType,
-	SequenceDeltaEvent,
-	SequenceEvent,
-	SequenceInterval,
-	SequenceMaintenanceEvent,
-	SharedIntervalCollection,
-	SharedIntervalCollectionFactory,
-	SharedSegmentSequence,
-	SharedSequence,
-	SharedString,
-	SharedStringFactory,
-	SubSequence,
-} from "@fluidframework/sequence";
-
-// The tree package manages its own API surface.
-// eslint-disable-next-line no-restricted-syntax
-export * from "@fluidframework/tree";
+	create,
+	DiscardResult,
+	disposeSymbol,
+	FieldKind,
+	FieldSchema,
+	IterableTreeListContent,
+	NodeKind,
+	RevertibleKind,
+	RevertResult,
+	SchemaFactory,
+	SharedTree,
```

Comment:
Kinda sucks that we have to publically expose APIs like summarize, getGCData etc here. Would be nice if at this level of abstraction, we could encapsulate that somehow,

Reply:
We expose SharedTree to be compatible with Odsp/AzureClient's "ContainerSchema", which requires a SharedObject subclass that exposes '.getFactory()'.

I have a beta 2 work item to avoid this so we can pass a () => IChannelFactory<ITree> (or similar) and remove the SharedTree export.